source/tutorial/triggers-fts.txt

```diff
@@ -0,0 +1,345 @@
+when a user creates a task that contains a specific word or phrase. The feature 
+includes:
+
+- An alert box using the :ref:`Realm React SDK <react-native-intro>` and 
+  :npm:`@realm/react <package/@realm/react>` that:
+
+  - Lets you enter the terms to be alerted on.
+  - Stores the specified terms in Atlas. 
+
+- A :ref:`Database Trigger <database-trigger>` with a custom
+  :ref:`Atlas Function <functions>` that:
+  
+  - Alerts you when a new task contains a term you've entered.
+  - Aggregates and returns related tasks by using an 
+    :atlas:`Atlas Search query </atlas-search/searching/>`.
+
+This example illustrates how you might use built-in features in Atlas  
+to add to your App Services application. You would not necessarily 
+make this change given the current structure of the template app.
+
+This tutorial should take around 30 minutes.
+
+Prerequisites
+-------------
+
+Before you begin: 
+
+- You must set up your local environment for React Native development. 
+  For detailed instructions, see `Setting up the development environment
+  <https://reactnative.dev/docs/environment-setup>`_ in the React Native
+  docs.
+
+.. include:: /includes/tutorial-template-prerequisite.rst 
+
+- To run Atlas Search queries, ensure that your Atlas cluster runs MongoDB 
+  version 4.2 or higher.
+
+Create a New Template App
+-------------------------
+
+This tutorial is based on the React Native SDK Flexible Sync Template App named 
+``react-native.todo.flex``. You start with the default app and build new features 
+on it. 
+
+To get the template app up and running on your computer, follow the
+steps described in the React Native tutorial: 
+
+1. :ref:`react-native-start-app` 
+
+#. :ref:`react-native-set-up-app`
+
+#. :ref:`react-native-get-familiar`
+
+Add an Alert Box 
+----------------
+
+After you've set up and explored the template app, write some code to 
+implement a new feature.
+
+In this section, you add an alert box that lets you enter terms that
+might appear in your important or time-sensitive tasks.
+
+.. procedure::
+
+   .. step:: Create the Alert Box Component
+
+      In the template app's ``src/`` directory, create a new ``AlertBox.tsx`` file 
+      and add the following code. This file contains the UI form that lets you 
+      enter the terms to be alerted on.
+
+      .. literalinclude:: /tutorial/backend-examples/AlertBox.tsx
+         :language: javascript
+         :caption: src/AlertBox.tsx
+
+   .. step:: Update the Main View
+
+      In ``src/ItemListView.tsx``, add the following import to the top of the file:
+
+      .. code-block:: javascript
+      
+         import {AlertBox} from './AlertBox';
+
+      Then, add the following lines to your code to render a button that 
+      displays the alert box when clicked:
+
+      .. literalinclude:: /tutorial/backend-examples/ItemListView.snippet.showAlertBox.tsx
+         :language: javascript
+         :caption: src/ItemListView.tsx
+         :emphasize-lines: 2
```

Comment:
A newbie react dev might be a bit confused as to where these two lines go. Can we expand the snippet to show a bit more context or at least give some helpful text? (e.g. you define them on the first and second line of the function block)